test/extended/builds/no_outputname.go

```diff
@@ -25,11 +25,21 @@ var _ = g.Describe("[builds][Conformance] build without output image", func() {
 			o.Expect(err).NotTo(o.HaveOccurred())
 
 			g.By("expecting build to pass without an output image reference specified")
-			out, err := oc.Run("start-build").Args("test-docker", "--follow", "--wait").Output()
+			err = oc.Run("start-build").Args("test-docker").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("waiting for the build to complete")
+			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), "test-docker-1", exutil.CheckBuildSuccessFn, exutil.CheckBuildFailedFn)
 			if err != nil {
-				fmt.Fprintln(g.GinkgoWriter, out)
+				exutil.DumpBuildLogs("test-docker", oc)
 			}
 			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("verifying the build test-docker-1 output")
+			out, err := oc.Run("logs").Args("-f", "bc/test-docker").Output()
+			fmt.Fprintf(g.GinkgoWriter, "\n: build logs: %s\n", out)
+			o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
@bparees The dumps have two different purposes. The first one dumps the logs if there is a build error and is present to ensure a consistent use of the requested pattern. The second retrieval/print is because the test's purpose is to search the output of the build logs. Even if the build is successful, the test could fail, and I wanted the developer to see the build log output in that event. 
Could it be moved into one pass? Absolutely. However, we would lose the pattern and I didn't want to do that until a real refactor/DRYing was practical.